test_data/makedata_suites/402_views.js

```diff
@@ -0,0 +1,470 @@
+      print(`making data 402 ${dbCount} ${loopCount}`);
+
+      // create analyzer with 'norm' feature
+      const analyzers = require("@arangodb/analyzers");
+      analyzers.save("AqlAnalyzerHash", "aql", { queryString: "return to_hex(to_string(@param))" }, ["frequency", "norm", "position"])
+
+      // create views for testing
+      progress('createViewCache');
+      let viewNameCache = `viewCache_${loopCount}`;
+      let viewCache = createSafe(viewNameCache,
+        viewNameCache => {
+          return db._createView(viewNameCache, "arangosearch", { "storedValues": [{ "fields": ["animal", "name"], "cache": true }] });
+        }, viewNameCache => {
+          return db._view(viewNameCache);
+        }
+      );
+
+      progress('createViewNoCache');
+      let viewNameNoCache = `viewNoCache_${loopCount}`;
+      let viewNoCache = createSafe(viewNameNoCache,
+        viewNameNoCache => {
+          return db._createView(viewNameNoCache, "arangosearch", { "storedValues": [{ "fields": ["animal", "name"], "cache": false }] });
+        }, viewNameNoCache => {
+          return db._view(viewNameNoCache);
+        }
+      );
+
+      let currVersion = db._version();
+      let cacheSizeSupported = isCacheSizeSupported(currVersion);
+
+      let cacheSize = 0;
+      let prevCacheSize = cacheSize;
+
+      if (cacheSizeSupported) {
+        cacheSize = getMetric("arangodb_search_columns_cache_size", options);
+        if (cacheSize != 0) {
+          throw new Error("initial cache size is not 0");
+        }
+        prevCacheSize = cacheSize;
+      }
+
+      for (let i = 0; i < links.length; i++) {
+        // create collection for each testing link
+        let collectionName = `collectionCache${i}_${loopCount}`;
+        createCollectionSafe(collectionName, 3, 1);
+        // insert some test data. Also insert version, on which 'make_data' was called
+        db._collection(collectionName).insert([
+          { "animal": "cat", "name": "tom" },
+          { "animal": "mouse", "name": "jerry" },
+          { "animal": "dog", "name": "harry" },
+          { "version": currVersion }
+        ]);
+
+        // add links to each created collection one by one
+        let meta = {
+          links: {}
+        };
+        meta.links[collectionName] = links[i];
+        viewCache.properties(meta);
+        viewNoCache.properties(meta);
+
+        if (cacheSizeSupported) {
+          // Should we check that current link will use cache?
+          let utilizeCache = links[i]["utilizeCache"]
+
+          // update cacheSize
+          cacheSize = getMetric("arangodb_search_columns_cache_size", options);
+          if ((cacheSize <= prevCacheSize) && utilizeCache) {
+            throw new Error("new cache size is wrong");
+          }
+          prevCacheSize = cacheSize;
+        }
+      }
+    },
+    checkData: function (options, isCluster, isEnterprise, dbCount, loopCount, readOnly) {
```

Comment:
shouldn't we in checkdata have something that iterates over all links and check their functionality?

Reply:
I have it:
```
            // current and previous versions are aware of 'cache'. 
            // Check that value is present and equal to value from previous version
            let expectedLink = links[i];
            if (!compareLinks(isCacheSupported, linkFromView, expectedLink)) {
              throw new Error(`links are not equal! ${linkFromView} ${expectedLink}`)
            }